src/extension.ts

```diff
@@ -58,15 +58,16 @@ export function activate(context: vscode.ExtensionContext) {
 	context.subscriptions.push(
 		vscode.workspace.onDidChangeConfiguration(
 			(event: vscode.ConfigurationChangeEvent) => {
-				if (!event.affectsConfiguration('terraform.languageServer')) {
+				if (event.affectsConfiguration("terraform")) {
+					const reloadMsg = "Reload VSCode window to apply language server changes";
+					vscode.window.showInformationMessage(reloadMsg, "Reload").then((selected) => {
+						if (selected === "Reload") {
+							vscode.commands.executeCommand("workbench.action.reloadWindow");
```

Comment:
Why does this need to restart, why can't we just use the configuration RPC hook?

Reply:
Just read the LS issue, this is fine if its just the initial implementation and we can improve it over time.

Reply:
@radeksimko and I talked through it this morning, this is the simplest option to start with. We'll have to stagger the language server and extension releases to deploy this, so it makes the most sense to do something we know we can ship on the language server quickly.

Reply:
The reload will still be necessary with the latest proposed solution (which involves `initializationOptions`), there's some context here to explain why:
https://github.com/hashicorp/terraform-ls/issues/189#issuecomment-652337667